internal/lm/resource.go

```diff
@@ -306,10 +307,10 @@ func getArchFamily(computeMajor, computeMinor int) string {
 
 func sanitise(input string) string {
 	var sanitised string
-	sanitised = strings.ReplaceAll(input, "(", "")
-	sanitised = strings.ReplaceAll(sanitised, ")", "")
+	re := regexp.MustCompile("[^a-zA-Z0-9-. ]")
```

Comment:
Could we use the same regex defined at https://github.com/kubernetes/apimachinery/blob/master/pkg/util/validation/validation.go#L31 ?

Reply:
I cant use that regex as is. 
API machinery regex is to identify unqualified label strings.
Regex I have added is to identify and replace the invalid characters.